src/lib/auth/pages/SignUpPage/SignUpPage.jsx

```diff
@@ -94,32 +100,30 @@ const SignUpPage = () => {
         }
         setLoader(false);
 
-        const authUser2 = await signupWithFirebase({
-          email: formValues.email,
-          password: formValues.password ?? '123456',
-          displayName: `${formValues.fullName} ${formValues.fullName}`,
+        await updateUser(newAuthUser, {
+          displayName: formValues.fullName?.trim(),
           photoURL: profilePicUrl,
         });
-        authUser = authUser2;
-        console.log({ authUser2 });
-
-        if (!authUser2) throw new Error('Failed to create user in firebase');
+        
+        authUser = newAuthUser;
+        if (!authUser) throw new Error('Failed to create user in firebase');
       }
+      
 
       console.log(authUser);
 
       const userDocData = new UserDoc({
-        first_name: formValues.fullName,
-        last_name: formValues.fullName,
-        card_id: formValues.id,
-        phone_number: formValues.phoneNumber,
-        role: roles.admin,
+        first_name: formValues.fullName.trim(),
+        last_name: formValues.fullName.trim(),
+        card_id: formValues.id.trim(),
+        phone_number: formValues.phoneNumber.trim(),
+        role: roles.applicant,
         appliciant_data: {
           field_of_study: formValues.fieldOfStudy,
           school_year: formValues.schoolYear,
           program: formValues.program,
           experience: formValues.experience,
-          experience_details: formValues.experienceDetails ?? '',
+          experience_details: (formValues.experienceDetails ?? '').trim(),
           test_day: sortingDates, //need to be an array of dates
```

Comment:
Will something like this work as well? :

`formValues.experienceDetails.trim() ?? '',`

Because there is no point triming empty string